sass/blocks/main-nav/_main-nav.scss

```diff
@@ -0,0 +1,90 @@
+.main-nav {
+  position: relative;
+  top: 0;
+  left: 0;
+  z-index: 1;
+  display: block;
+  width: 100%;
+  font-weight: $font-bolded;
+  text-align: center;
+
+  &__list {
+    position: relative;
+    z-index: 2;
+    display: block;
+    width: 100%;
+    margin: 0;
+    padding: 0;
+    background: $color-prmiary;
+  }
+
+  &__item {
+    position: relative;
+    display: block;
+    height: 5.4rem;
+    line-height: 5.4rem;
+    list-style-type: none;
+    border-top: 0.1rem solid $top-border;
+    border-bottom: 0.1rem solid $bottom-border;
+
+    & > a {
+      display: block;
+      font-size: $font-size--nav;
+      text-transform: uppercase;
+
+      &:hover {
+        background-color: $color-hover-btn;
+      }
+    }
+
+    &:first-child {
+      border-top: 0;
+    }
+
+    &:last-child {
+      height: 5.5rem;
+      border-bottom: 0;
+    }
+  }
+
+  &__toggle {
+    position: absolute;
+    top: 1.2rem;
+    right: 1.7rem;
+    z-index: 2;
```

Comment:
I can see sprinkles of z-index across this file. Don't know why you need it. The only element that potentially needs z-index here is the logo, I would start from there